### PR TITLE
Unify workloadentry and pod topology label

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

This is an improvement after #34487

FYI: https://github.com/istio/istio/pull/34487#discussion_r684020982

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/serviceregistry/kube/controller/multicluster.go [hunk 1]
```diff
@@ -197,12 +197,13 @@ func (m *Multicluster) AddMemberCluster(clusterID cluster.ID, rc *secretcontroll
 		} else if features.WorkloadEntryCrossCluster {
 			// TODO only do this for non-remotes, can't guarantee CRDs in remotes (depends on https://github.com/istio/istio/pull/29824)
 			if configStore, err := createConfigStore(client, m.revision, options); err == nil {
-				m.remoteKubeControllers[clusterID].workloadEntryStore = serviceentry.NewServiceDiscovery(
+				m.remoteKubeControllers[clusterID].serviceEntryStore = serviceentry.NewServiceDiscovery(
```
Comment (stevenctl) [inline pilot/pkg/serviceregistry/kube/controller/multicluster.go:200]: nit: while the type is technically serviceentry store, the name helps make it clear that this one only processes WorkloadEntry.

**Review by stevenctl — APPROVED** [review]

**Comment by stevenctl** [comment]
LGTM! Feel free to resolve the nit or just remove the hold and merge. 

In the future we might split this to a `NetworksController` component that is shared between the k8s controller and service entry controller.

File: pilot/pkg/serviceregistry/kube/controller/multicluster.go (hunk 1 above)
Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/kube/controller/multicluster.go]: ok, will revert the name unless once we support other types

**Pull request merged**